opensearch/Dockerfile

```diff
@@ -1,7 +1,27 @@
 # Repo name: arynai/sycamore-opensearch
 
 ARG TAG=2.12.0
-FROM opensearchproject/opensearch:$TAG
+
+FROM gradle:jdk17 as build_ml_commons
+
+WORKDIR /build_ml
+RUN git clone https://github.com/opensearch-project/ml-commons.git
+WORKDIR /build_ml/ml-commons
+RUN git checkout -b 2.12 origin/2.12
+RUN sed -i 's#if (os.macOsX) {#if (System.getProperty("os.arch") == "aarch64") {#' ml-algorithms/build.gradle
+RUN ./gradlew assemble
```

Comment:
If we're not doing the other sed, then for amd64 the plugin is unchanged.  Given that, I'd recommend:
```
RUN /mkdir /tmp/release
RUN [[ $(uname -m) == arm64 ]] && cp /build_ml/ml-commons/plugin/build/distributions/opensearch-ml-2.12.0.0-SNAPSHOT.zip /tmp/release
# uname -m is a guess, might be wrong

# then below
COPY --chown=opensearch:opensearch --from=build_ml_commons /tmp/release /tmp
COPY maybe-update-mlcommons.sh /tmp
```
with maybe-update-mlcommons.sh:
#!/bin/sh
if [[ ! -f /tmp/opensearch-ml-2.12.0.0-SNAPSHOT.zip ]]; then
   echo "No opensearch plugin to install"
   exit 0
fi
... rest of the install steps
```

Reply:
looking like the other sed is valuable, so we'll use it. Even though it makes me very incomfortable